### DISCUSSION TITLE
Ask about full document content in chat

### DISCUSSION THREAD

**Discussion by dmvieira** [open-webui/open-webui#5508] [Ideas]
**Is your feature request related to a problem? Please describe.**
When I upload documents is impossible to ask for summarization of them.

**Describe the solution you'd like**
If we include a document in chat, we should be able to summarize it and correlate with other documents previews added in chat.

**Describe alternatives you've considered**
If we include a document in chat, we should use all the document content in prompt instead just RAG on it

**Comment by thiswillbeyourgithub** [comment]
I think the best way would be to give the choice to the user for each document or depending on the size. Several possibilities

1. Add a setting to specify the token threshold for which we switch from putting the whole content to only the topk sections. I don't like this because as model vary wildly in capabilities it should be a per model setting in theory. It can also depend on the type of document and the situation.
2. Duplicate the ui: a button to send the file as a whole, and another to use RAG. This would clutter the UI
3. Add a mark at the top right corner of the file in the text field. This mark can be clicked to switch from RAG to whole document and visually changes according to the selected mode. This seems best to me. In addition we could add a setting to set the threshold and the default value (rag or whole) to change the default mark value on upload. I would be fine with this setting being system wide instead of per model. The best being a system wide + a per model override.
4. Adding a button on the top right corner of a file to switch to whole mode : the file icon would disappear and instead the chatbox would be filled with the parsed content. This would make it pretty intuitive as "whole text" would be simply passed inside the user message

**Reply by dmvieira** [reply]
3

**Comment by dmvieira** [comment]
Option 3 sounds really good!

**Comment by dmvieira** [comment]
I found here a project that can help with this file memory that's enable bigger files: https://github.com/qhjqhj00/MemoRAG

**Comment by tjbck** [comment]
https://github.com/open-webui/open-webui/issues/3129#issuecomment-2381599443
